web/source/kmwbase.ts

```diff
@@ -176,15 +176,26 @@ namespace com.keyman {
       this.touchAliasing = this.util.device.touchable ? this.domManager.touchHandlers : this.domManager.nonTouchHandlers;
     }
 
+    /**
+     * Reset context when entering or exiting the active element.
+     * Will also trigger OSK shift state / layer reset.
+     **/
+    pageFocusHandler: () => boolean = function(this: KeymanBase) {
+      if(!this.uiManager.isActivating && this.osk?.vkbd) {
+        this.core.resetContext(null);
+      }
+      return false;
+    }.bind(this);
```

Comment:
Yep, looks like it was pretty clean to move this.

```suggestion
    pageFocusHandler: () => boolean = () => {
      if(!this.uiManager.isActivating && this.osk?.vkbd) {
        this.core.resetContext(null);
      }
      return false;
    };
```

Query, do we need the `bind` pattern here if we use an arrow function which has lexical `this`?

One step further, can Typescript infer the type of `pageFocusHandler`?

```suggestion
    pageFocusHandler = () => {
      if(!this.uiManager.isActivating && this.osk?.vkbd) {
        this.core.resetContext(null);
      }
      return false;
    };
```

Reply:
![image](https://user-images.githubusercontent.com/25213402/178868994-ae8440df-1b51-418d-b539-1879acf7b5ed.png)

Infers it well enough.